contracts/stellar-axelar-gateway/src/storage.rs

```diff
@@ -15,17 +16,42 @@ pub enum MessageApprovalValue {
     Executed,
 }
 
-#[contracttype]
+#[contractstorage]
 #[derive(Clone, Debug)]
 pub enum DataKey {
     /// Gateway
-    MessageApproval(MessageApprovalKey),
+
+    #[persistent]
+    #[value(MessageApprovalValue)]
+    MessageApproval { key: MessageApprovalKey },
```

Comment:
Create a ticket to remove this wrapper key type altogether, we can use the fields directly. This PR should preserve the existing schema but we can make it cleaner in a follow up with a potential migration